networkx/algorithms/summarization.py

```diff
@@ -0,0 +1,219 @@
+        >>> original_graph.number_of_edges()
+        15
+        >>> c_graph.number_of_edges()
+        14
+
+    A dedensified, directed graph can be "densified" to reconstruct the
+    original graph::
+
+        >>> original_graph = nx.DiGraph()
+        >>> original_graph.add_nodes_from(
+        ...     ["1", "2", "3", "4", "5", "6", "A", "B", "C"]
+        ... )
+        >>> original_graph.add_edges_from(
+        ...     [
+        ...         ("1", "C"), ("1", "B"),
+        ...         ("2", "C"), ("2", "B"), ("2", "A"),
+        ...         ("3", "B"), ("3", "A"), ("3", "6"),
+        ...         ("4", "C"), ("4", "B"), ("4", "A"),
+        ...         ("5", "B"), ("5", "A"),
+        ...         ("6", "5"),
+        ...         ("A", "6")
+        ...     ]
+        ... )
+        >>> c_graph, c_nodes = nx.dedensify(original_graph)
+        >>> for c_node in c_nodes:
+        ...     all_neighbors = set(nx.all_neighbors(c_graph, c_node))
+        ...     out_neighbors = set(c_graph.neighbors(c_node))
+        ...     for out_neighbor in out_neighbors:
+        ...         c_graph.remove_edge(c_node, out_neighbor)
+        ...     in_neighbors = all_neighbors - out_neighbors
+        ...     for in_neighbor in in_neighbors:
+        ...         c_graph.remove_edge(in_neighbor, c_node)
+        ...         for out_neighbor in out_neighbors:
+        ...             c_graph.add_edge(in_neighbor, out_neighbor)
+        ...     c_graph.remove_node(c_node)
+        ...
+        >>> nx.is_isomorphic(original_graph, c_graph)
+        True
+
+    References
+    ----------
+    .. [1] Maccioni, A., & Abadi, D. J. (2016, August).
+       Scalable pattern matching over compressed graphs via dedensification.
+       In Proceedings of the 22nd ACM SIGKDD International Conference on
+       Knowledge Discovery and Data Mining (pp. 1755-1764).
+       http://www.cs.umd.edu/~abadi/papers/graph-dedense.pdf
+    """
+    high_degree_nodes = set()
+    low_degree_nodes = set()
+    for node in G.nodes():
+        if isinstance(G, nx.DiGraph):
+            node_degrees = G.in_degree(node)
+        else:
+            node_degrees = G.degree(node)
+        if node_degrees > threshold:
+            high_degree_nodes.add(node)
+        else:
+            low_degree_nodes.add(node)
+
+    auxillary = dict()
+    for node in high_degree_nodes | low_degree_nodes:
+        node_neighbors = set(G.neighbors(node))
+        high_degree_neighbors = frozenset(node_neighbors & high_degree_nodes)
+        if high_degree_neighbors:
+            low_degree_neighbors = auxillary.get(high_degree_neighbors)
+            if not low_degree_neighbors:
+                auxillary[high_degree_neighbors] = set([node])
+            else:
+                low_degree_neighbors.add(node)
+
+    if not inplace:
+        G = G.copy()
+
+    compressor_nodes = set()
+    for index, nodes in enumerate(auxillary.items()):
+        high_degree_nodes, low_degree_nodes = nodes
```

Comment:
```suggestion
    for index, (high_degree_nodes, low_degree_nodes)  in enumerate(auxillary.items()):
```

Reply:
@dschult  I like how this looks and its actually how I originally had written it, but it will bring the line length beyond 80 characters which would raise PEP8 issues.  If those can be ignored, I'll apply this change

Reply:
We use black for our style and it uses 88 char line lengths.  You can learn more about our recommended setup here:
https://networkx.org/documentation/latest/developer/contribute.html

To run black manually
```
pip install -r requirements/developer.txt
black <file>
```
or you can set up the pre-commit hook as we recommend in the developer guide.

Reply:
We recently disabled pep8speaks, but if you see it complaining just ignore it.